projects/batfish-common-protocol/src/main/java/org/batfish/datamodel/PrefixRange.java

```diff
@@ -2,78 +2,50 @@
 
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonValue;
+import java.io.Serializable;
+import java.util.Objects;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 import org.batfish.common.BatfishException;
-import org.batfish.common.Pair;
 
-public class PrefixRange extends Pair<Prefix, SubRange> {
+public final class PrefixRange implements Serializable, Comparable<PrefixRange> {
 
   /** */
   private static final long serialVersionUID = 1L;
 
+  public PrefixRange(Prefix prefix, SubRange lengthRange) {
+    // Canonicalize the prefix by dropping extra bits in the address that are longer than any
+    // relevant length.
+    int realPrefixLength = Math.min(prefix.getPrefixLength(), lengthRange.getEnd());
```

Comment:
there is an open issue here as to whether this is precisely what's desired. @arifogel